lib/cloudregister/registerutils.py

```diff
@@ -516,6 +517,90 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    """Set all the necessary parts for the registry,
+       returns True if the setup completed, False otherwise."""
+    if not os.path.exists(os.path.dirname(REGISTRY_CREDENTIALS_PATH)):
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH))
+
+    setup_registry_succeed = set_registry_auth_token(
+        registry_fqdn, username, password
+    )
+    return setup_registry_succeed
+
+
+# ----------------------------------------------------------------------------
+def get_registry_credentials():
+    """Read the registry credentials file
+       and return its content or an empty dict."""
+    config_json = {}
+    failed = False
+    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        try:
+            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+                config_json = json.load(cred_json)
+        except json.decoder.JSONDecodeError:
+            logging.info(
+                'Unable to parse existing %s, preserving file as %s.bak, '
+                'writing new credentials' % (
+                    REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+                )
+            )
+            # we want to remove the possible
+            # auth token present in the file
+            # before preserving it
+            mv_file_cmd = 'mv -Z {} {}.bak'.format(
+                REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+            ).split()
+            failed = exec_subprocess(mv_file_cmd)
+            message = 'File not preserved.' if failed else 'File preserved.'
+            logging.info(message)
+
+    return config_json, failed
+
+
+# ----------------------------------------------------------------------------
+def write_registry_credentials(content):
+    """Update the registry credentials file with the value of 'content'."""
+    try:
+        with open(REGISTRY_CREDENTIALS_PATH, 'w') as cred_json_file:
+            json.dump(content, cred_json_file)
+        logging.info(
+            'Credentials for the registry set in %s' %
+            REGISTRY_CREDENTIALS_PATH
+        )
+        return True
+    except Exception as error:
```

Comment:
we catch any Exception, intentionally ?

Reply:
Yes, if we cannot write the registry credentials, for whatever reason we still want to proceed and at least preserve the  registration of the repositories.

Reply:
> if we cannot write the registry credentials, for whatever reason we still want to proceed and at least preserve the registration of the repositories.

I agree with that, but the "whatever reason" part could be more specific, right ?
either the file fails to open or the dump failed, do we want to capture a general `Exception` ?

Reply:
I do not think we need to be that pedantic and catching anything that shows up at this point here is fine.